test/Dialect/FIRRTL/lower-layers.mlir

```diff
@@ -471,3 +471,82 @@ firrtl.circuit "ModuleNameConflict" {
+    firrtl.layerblock @A {
+      %0 = firrtl.add %c0_ui1, %c0_ui1 : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<2>
+    }
+  }
+
+  // CHECK: firrtl.module private @[[A:.+]](out %[[dst:.+]]: !firrtl.probe<uint<1>>, in %[[src:.+]]: !firrtl.uint<1>)
+  // CHECK:   %0 = firrtl.ref.send %[[src]] : !firrtl.uint<1>
+  // CHECK:   %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+  // CHECK:   %1 = firrtl.ref.send %c0_ui1 : !firrtl.uint<1>
+  // CHECK:   firrtl.ref.define %[[dst]], %1 : !firrtl.probe<uint<1>>
+  // CHECK:   %2 = firrtl.ref.resolve %0 : !firrtl.probe<uint<1>>
+  // CHECK: }
+  // CHECK: firrtl.module @CaptureAsDstThenSrc() {
+  // CHECK:   %w = firrtl.wire : !firrtl.probe<uint<1>>
+  // CHECK:   %[[out:.+]], %[[in:.+]] = firrtl.instance {{.+}} @[[A]](out {{.+}}: !firrtl.probe<uint<1>>, in {{.+}}: !firrtl.uint<1>)
+  // CHECK:   %0 = firrtl.ref.resolve %w : !firrtl.probe<uint<1>>
+  // CHECK:   firrtl.strictconnect %[[in]], %0 : !firrtl.uint<1>
+  // CHECK:   firrtl.ref.define %w, %[[out]] : !firrtl.probe<uint<1>>
+  // CHECK: }
+  firrtl.module @CaptureAsDstThenSrc() {
+    %w = firrtl.wire : !firrtl.probe<uint<1>, @A>
+    firrtl.layerblock @A {
+      // capture first as a sink.
+      %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+      %0 = firrtl.ref.send %c0_ui1 : !firrtl.uint<1>
+      %1 = firrtl.ref.cast %0 : (!firrtl.probe<uint<1>>) -> !firrtl.probe<uint<1>, @A>
+      firrtl.ref.define %w, %1 : !firrtl.probe<uint<1>, @A>
+      // Capture again, as a source.
+      %2 = firrtl.ref.resolve %w : !firrtl.probe<uint<1>, @A>
+    }
+  }
+
+  // CHECK: firrtl.module private @[[A:.+]](in %[[src:.+]]: !firrtl.uint<1>, out %[[dst:.+]]: !firrtl.probe<uint<1>>)
+  // CHECK:   %0 = firrtl.ref.send %[[src]] : !firrtl.uint<1>
+  // CHECK:   %1 = firrtl.ref.resolve %0 : !firrtl.probe<uint<1>>
+  // CHECK:   %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+  // CHECK:   %2 = firrtl.ref.send %c0_ui1 : !firrtl.uint<1>
+  // CHECK:   firrtl.ref.define %[[dst]], %2 : !firrtl.probe<uint<1>>
+  // CHECK: }
+  // CHECK: firrtl.module @CaptureAsSrcThenDst() {
+  // CHECK:   %w = firrtl.wire : !firrtl.probe<uint<1>>
+  // CHECK:   %[[in:.+]], %[[out:.+]] = firrtl.instance {{.+}} @[[A]]
+  // CHECK:   firrtl.ref.define %w, %[[out]] : !firrtl.probe<uint<1>>
+  // CHECK:   %0 = firrtl.ref.resolve %w : !firrtl.probe<uint<1>>
+  // CHECK:   firrtl.strictconnect %[[in]], %0 : !firrtl.uint<1>
+  // CHECK: }
+  firrtl.module @CaptureAsSrcThenDst() {
+    %w = firrtl.wire : !firrtl.probe<uint<1>, @A>
+    firrtl.layerblock @A {
+      // capture first as a source.
+      %2 = firrtl.ref.resolve %w : !firrtl.probe<uint<1>, @A>
+      // capture again, as a sink.
+      %c0_ui1 = firrtl.constant 0 : !firrtl.uint<1>
+      %0 = firrtl.ref.send %c0_ui1 : !firrtl.uint<1>
+      %1 = firrtl.ref.cast %0 : (!firrtl.probe<uint<1>>) -> !firrtl.probe<uint<1>, @A>
+      firrtl.ref.define %w, %1 : !firrtl.probe<uint<1>, @A>
+    }
+  }
+}
```

Comment:
Nits in new tests: 

The type information is really unimportant for these tests. Consider dropping it where unnecessary as it doesn't aid test readability.